components/toast/script.js

```diff
@@ -0,0 +1,174 @@
+ */
+class GamefaceToast extends BaseComponent {
+    /* eslint-disable require-jsdoc */
+    constructor() {
+        super();
+        this.template = template;
+        this.init = this.init.bind(this);
+        this.hide = this.hide.bind(this);
+        this.show = this.show.bind(this);
+        this.hideTimeOut = null;
+        this._gravity = 'top';
+        this._position = 'left';
+    }
+
+    get message() {
+        return this._messageSlot.textContent;
+    }
+
+    set message(value) {
+        this._messageSlot.innerHTML = value;
+    }
+
+    get targetElement() {
+        return this._targetElement;
+    }
+
+    set targetElement(element) {
+        this._targetElement = element;
+    }
+
+    get position() {
+        return this._position;
+    }
+
+    set position(value) {
+        this._position = value;
+        this.setAttribute('position', value);
+    }
+
+    get gravity() {
+        return this._gravity;
+    }
+
+    set gravity(value) {
+        this._gravity = value;
+        this.setAttribute('gravity', value);
+    }
+
+    init(data) {
+        this.setupTemplate(data, () => {
+            components.renderOnce(this);
+            // attach event handlers here
+            this.attachEventListeners();
+            this._messageSlot = this.querySelector('.guic-toast-message').firstElementChild;
+        });
+    }
+
+    connectedCallback() {
+        this.gravity = this.getAttribute('gravity') || this._gravity;
+        this.position = this.getAttribute('position') || this._position;
+        this.timeout = this.getAttribute('timeout') || 0;
+        this.elementSelector = this.getAttribute('target');
+        this._targetElement = this._targetElement || document.querySelector(this.elementSelector);
+
+        if (!containersCreated) this.createToastContainers();
+
+        components.loadResource(this)
+            .then(this.init)
+            .catch(err => console.error(err));
+    }
+
+    attachEventListeners() {
+        if (this._targetElement ) {
+            this._targetElement.addEventListener('click', this.show);
+        }
+
+        if (!animationEventAttached) {
+            document.addEventListener('animationend', (event) => {
+                if (event.animationName === 'guic-toast-fade-out') event.target.parentElement.removeChild(event.target);
+            });
+            animationEventAttached = true;
+        }
+    }
+    /* eslint-enable require-jsdoc */
+
+    /**
+     * Creates containers for all possible toast positions
+    */
+    createToastContainers() {
+        const body = document.querySelector('body');
+        TOAST_POSITIONS.forEach((containerPosition) => {
+            const [vertical, horizontal] = containerPosition.split(' ');
+            const toastContainer = document.createElement('div');
+            toastContainer.classList.add('guic-toast-container', `${classPrefix}${vertical}`, `${classPrefix}${horizontal}`);
+            body.appendChild(toastContainer);
```

Comment:
Try to add to the containers `pointer-events: none` and to the toasts `pointer-events: auto` to see if the containers will pass mouse events to the elements bellow. Ask for more info

Reply:
Fixed